hnn_core/viz.py

```diff
@@ -64,21 +77,69 @@ def plot_hist_input(net, ax=None, show=True):
         The matplotlib figure handle.
     """
     import matplotlib.pyplot as plt
-    spikes = np.array(sum(net.spikes.times, []))
-    gids = np.array(sum(net.spikes.gids, []))
-    valid_gids = np.r_[[v for (k, v) in net.gid_dict.items()
-                        if k.startswith('evprox')]]
-    mask_evprox = np.in1d(gids, valid_gids)
-    valid_gids = np.r_[[v for (k, v) in net.gid_dict.items()
-                        if k.startswith('evdist')]]
-    mask_evdist = np.in1d(gids, valid_gids)
-    bins = np.linspace(0, net.params['tstop'], 50)
+    spike_times = np.array(sum(spikes._times, []))
+    spike_types_data = np.array(sum(spikes._types, []))
+
+    unique_types = np.unique(spike_types_data)
+    spike_types_mask = {s_type: np.in1d(spike_types_data, s_type)
+                        for s_type in unique_types}
+    cell_types = ['L5_pyramidal', 'L5_basket', 'L2_pyramidal', 'L2_basket']
+    input_types = np.setdiff1d(unique_types, cell_types)
+
+    if isinstance(spike_types, str):
+        spike_types = {spike_types: [spike_types]}
+
+    if spike_types is None:
+        if any(input_types):
+            spike_types = input_types.tolist()
+        else:
+            spike_types = unique_types.tolist()
+    if isinstance(spike_types, list):
+        spike_types = {s_type: [s_type] for s_type in spike_types}
+    if isinstance(spike_types, dict):
+        for spike_label in spike_types:
+            if not isinstance(spike_types[spike_label], list):
+                raise TypeError(f'spike_types[{spike_label}] must be a list. '
+                                f'Got '
+                                f'{type(spike_types[spike_label]).__name__}.')
```

Comment:
```suggestion
                                f'{type(spike_types[spike_label])')
```

do you need the "name"?

Reply:
Yeah while `print(type(int))` seems to convert type objects to strings just fine, it seems to be omitted when included in an error message. (at least on my computer).